### PR TITLE
Prevent profession items from being auto worn at character creation.

### PR DESCRIPTION
**Pull request by OzoneH3**
Implements #16624

This is working but it feels kinda hackish.

### CODE REVIEW

File: src/item_group.cpp [hunk 1]
```diff
@@ -265,10 +264,17 @@ void Item_modifier::modify(item &new_item) const
             new_item = cont;
         }
     }
+
     if (contents.get() != NULL) {
         Item_spawn_data::ItemList contentitems = contents->create(new_item.bday);
         new_item.contents.insert(new_item.contents.end(), contentitems.begin(), contentitems.end());
     }
+
+    if( !custom_flags.empty() ) {
```
Comment (BevapDin) [inline src/item_group.cpp]: This check is pointless. The following loop will simply do nothing if `custom_flags` is empty.

**Comment by Coolthulhu** [comment]
Why not keep unworn items as a separate vector, for example "inventory"?
It could be loaded just like the worn items, just applied differently.

**Comment by OzoneH3** [comment]
> Why not keep unworn items as a separate vector, for example "inventory"?
It could be loaded just like the worn items, just applied differently.

Not exactly sure where I'd put that.
Add another vector besides ```typedef std::vector<item> ItemList;``` in ```class Item_group : public Item_spawn_data``` ?

**Comment by Coolthulhu** [comment]
> Not exactly sure where I'd put that.

Each profession has an item group for its worn items for both, male and female. You could add a second item group just for un-worn items.
Though I'm not sure if it's better than marking items as "don't wear".

**Comment by OzoneH3** [comment]
Yeah, at least the flags could potentially be used for something else as well in the future.

**Comment by BorkBorkGoesTheCode** [comment]
The go bag is going, going, gone!

**Pull request merged**